cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-s2i-deployments.adoc

```diff
@@ -0,0 +1,221 @@
++
+The example emulates an HAProxy config file, and is typically used for overriding default configurations in an OpenShift application. Files can even be renamed in the ConfigMap.
++
+In your CLI enter the following command, replacing `<UserName>` with your GitHub username:
++
+[source,terminal]
+----
+$ oc create -f https://raw.githubusercontent.com/<UserName>/ostoy/master/deployment/yaml/configmap.yaml
+----
+
+. Deploy the microservice
++
+You must deploy the microservice first to ensure that the SERVICE environment variables are available from the UI application. `--context-dir` is used here to only build the application defined in the `microservice` directory in the git repository. Using the `app` label allows us to ensure the UI application and microservice are both grouped in the OpenShift UI. Run the following command in the CLI to create the microservice, replacing `<UserName>` with your GitHub username::
++
+[source,terminal]
+----
+$ oc new-app https://github.com/<UserName>/ostoy \
+    --context-dir=microservice \
+    --name=ostoy-microservice \
+    --labels=app=ostoy
+---- 
++
+.Example output
+[source,terminal]
+----
+Creating resources with label app=ostoy ...
+  imagestream "ostoy-microservice" created
+  buildconfig "ostoy-microservice" created
+  deploymentconfig "ostoy-microservice" created
+  service "ostoy-microservice" created
+Success
+  Build scheduled, use 'oc logs -f bc/ostoy-microservice' to track its progress.
+  Application is not exposed. You can expose services to the outside world by executing one or more of the commands below:
+    'oc expose svc/ostoy-microservice'
+  Run 'oc status' to view your app.
+----
+
+. Check the status of the microservice
++
+Before moving onto the next step we should be sure that the microservice was created and is running correctly by running the following command:
++
+[source,terminal]
+----
+$ oc status
+----
++
+.Example output
+[source,terminal]
+----
+In project ostoy-s2i on server https://api.myrosacluster.g14t.p1.openshiftapps.com:6443
+
+svc/ostoy-microservice - 172.30.47.74:8080
+  dc/ostoy-microservice deploys istag/ostoy-microservice:latest <-
+    bc/ostoy-microservice source builds https://github.com/0kashi/ostoy on openshift/nodejs:14-ubi8
+    deployment #1 deployed 34 seconds ago - 1 pod
+----
++
+Wait until you see that it was successfully deployed. You can also check this through the web UI.
+
+. Deploy the front end UI
++
+The application has been designed to rely on several environment variables to define external settings. Attach the previously created Secret and ConfigMap afterward, along with creating a PersistentVolume. Enter the following into the CLI:
++
+[source,terminal]
+----
+$ oc new-app https://github.com/<UserName>/ostoy \
+    --env=MICROSERVICE_NAME=OSTOY_MICROSERVICE
+----
++
+.Example output
++
+[source,terminal]
+----
+Creating resources ...
+   imagestream "ostoy" created
+   buildconfig "ostoy" created
+   deploymentconfig "ostoy" created
```

Comment:
same here, this is not a deploymentconfig, the line is: "deployment.apps "ostoy" created"